src/azure-cli/azure/cli/command_modules/config/_params.py

```diff
@@ -11,20 +11,27 @@ def load_arguments(self, _):
         c.ignore('_subscription')  # ignore the global subscription param
 
     with self.argument_context('config set') as c:
-        c.positional('key_value', nargs='+', help="Space-separated configurations in the form of <section>.<key>=<value>.")
+        c.positional('key_value_pairs', nargs='+',
+                     help="Set options in the form of <section>.<name>=<value>. "
+                          "If `<section>` is omitted, `core` will be used. "
```

Comment:
It's a little strange to me that the omitted value is not `defaults`.

Reply:
If we default section to `defaults`, setting options in `core` section will still be

```
az config set core.output=table
```

This is not very user-friendly. 

On the other hand, setting the default group becomes:

```
az config set group=rg1
```

which may also confuse users as some commands ignore the default resource group, like `az role assignment create`.

https://github.com/Azure/azure-cli/blob/37d15261c516bc095236ddb1155dbcc470daf79b/src/azure-cli/azure/cli/command_modules/role/_validators.py#L98-L101

Also, defaulting section to `core` is also what gcloud adopts in [`gcloud config set`](https://cloud.google.com/sdk/gcloud/reference/config/set):

> SECTION/PROPERTY
> Property to be set. Note that SECTION/ is optional while referring to properties in the core section, i.e., using either core/project or project is a valid way of setting a project. Using section names is required for setting other properties like compute/region. Consult the Available Properties section below for a comprehensive list of properties.